src/main/java/hudson/plugins/ec2/util/AmazonEC2Factory.java

```diff
@@ -0,0 +1,29 @@
+package hudson.plugins.ec2.util;
+
+import java.net.URL;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.ec2.AmazonEC2;
+import hudson.ExtensionPoint;
+import jenkins.model.Jenkins;
+
+public abstract class AmazonEC2Factory implements ExtensionPoint {
```

Comment:
Just change this to an interface instead of an abstract class and we're good to go!

Reply:
Fixed now. Also did some cleanup and fixed my commit author since I did this work from two different computers.

Reply:
Should I squish the commits too?

Reply:
squash*

Reply:
Yes, go ahead.